system/HTTP/CURLRequest.php

```diff
@@ -705,9 +707,16 @@ protected function setCURLOptions(array $curlOptions = [], array $config = [])
         }
 
         // Cookie
-        if (isset($config['cookie'])) {
-            $curlOptions[CURLOPT_COOKIEJAR]  = $config['cookie'];
-            $curlOptions[CURLOPT_COOKIEFILE] = $config['cookie'];
+        if ($cookie = ($config['cookie_jar'] ?? $config['cookie'] ?? false)) {
+            $curlOptions[CURLOPT_COOKIEJAR] = $cookie;
+        }
+
+        if ($cookie = ($config['cookie_file'] ?? $config['cookie'] ?? false)) {
+            $curlOptions[CURLOPT_COOKIEFILE] = $cookie;
+        }
+
+        if ($cookie = ($config['cookie_session'] ?? false)) {
+            $curlOptions[CURLOPT_COOKIESESSION] = $cookie;
```

Comment:
I understand the unusual structure above but this one can use the typical format:
```suggestion
        if (isset($config['cookie_session'])) {
            $curlOptions[CURLOPT_COOKIESESSION] = $config['cookie_session'];
```

Reply:
In this case there should be only boolean value passed. I will update it according to your coding style advice.